src/engine/filters/enginefilterdelay.h

```diff
@@ -29,6 +29,13 @@ class EngineFilterDelay : public EngineObjectConstIn {
 
     void setDelay(unsigned int delaySamples) {
         m_delaySamples = delaySamples;
+
+        // When mixxx will support other channel count variants than stereo,
+        // the kMaxDelay has to be divisible by the number of channels.
+        // Otherwise, channels may be swapped.
+        VERIFY_OR_DEBUG_ASSERT(m_delaySamples <= SIZE - mixxx::kEngineChannelCount) {
+            m_delaySamples = SIZE - mixxx::kEngineChannelCount;
```

Comment:
The original code just performs a. copy.
(Delay = 0) in that case. Since we are anyway In an anti crash backup path, I find mind, I just want to pint it out.

Reply:
I agree, that's right. This PR just offers another point of view for the problem, that the wrong delay values can be clamped in the setter. This version should be anti-crash too. Of course, I will be okay, if the PR won't be merged. It is just proposed to use a more similar approach as in [EngineDelay](https://github.com/mixxxdj/mixxx/blob/f2d670edeb483efb066bf7f00255d01df2f0017c/src/engine/enginedelay.cpp) or the future used `EngineEffectsDelay`.